modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -36,8 +36,10 @@ export type TrainInput = Readonly<{
   contexts: string[]
   intents: Intent<string>[]
   trainingSession: TrainingSession
+  ctxToTrain: string[]
```

Comment:
I find it a bit confusing to pass a `ctxToTrain` it seems to a piece of logic that is proper to engine and previous training sessions which should be completely hidden from the training pipeline.

I think the same could be achieved by either removing intents from the inputs or by removing the already trained ctx from intents. That will leave the training pipeline logic free

Reply:
I completely agree, but I don't think it can be implemented differently.

The context classifier needs all contexts to train.